internal/tools/exechelper/run.go

```diff
@@ -54,6 +55,12 @@ func Run(cmd *exec.Cmd, logger log.FieldLogger, outputLogger OutputLogger) ([]by
 		"args": cmd.Args,
 	}).Info("Invoking command")
 
+	expandedArgs := []string{}
+	for _, arg := range cmd.Args {
+		expandedArgs = append(expandedArgs, os.ExpandEnv(arg))
+	}
+	cmd.Args = expandedArgs
```

Comment:
I am a bit concerned about this as we discussed in chat. Due to this affecting all commands, it seems slightly risky in regards to introducing unwanted behavior somewhere that would be harder to track down.

I feel like we can change the helm logger logic to avoid this so I would advocate for trying that if possible.

Reply:
The line that needs to be censored is actually logged out by the regular log handler and not the Helm logger, it's the line that starts with `Invoking command` just above this hunk. But I will look to see if I can move some things around and introduce a `RunWithEnvironmentVariables` that does the same thing as `Run` but expands env vars like we need for this operation, without potentially affecting all of the other call sites for `Run`.

Reply:
Potential resolution to this thread in 4f5a42f